android/MLCChat/mlcengineexample/src/main/java/ai/mlc/mlcengineexample/MLCEngine.kt

```diff
@@ -0,0 +1,40 @@
+package ai.mlc.mlcengineexample
+
+import ai.mlc.mlcllm.JSONFFIEngine
+import android.util.Log
+import kotlin.concurrent.thread
+
+class MLCEngine (private val streamCallback: (String) -> Unit) {
```

Comment:
Move MLCEngine to be be part of the mlc4j library (implement in java?), JSONFFIEngine should not be exposed to the users, and should be used as internal. Users should interact with MLCEngine. 

(this can be a TODO)  Ideally, we should not directly use stream callback, but instead put the requests into queues in stream callback, then allow chatCompletion to iterate over these queues.